tests/e2e/utils/workspace/TestWorkspaceUtil.ts

```diff
@@ -33,31 +35,99 @@ export class TestWorkspaceUtil implements ITestWorkspaceUtil {
     public async waitWorkspaceStatus(namespace: string, workspaceName: string, expectedWorkspaceStatus: WorkspaceStatus) {
         Logger.debug('TestWorkspaceUtil.waitWorkspaceStatus');
 
-        const workspaceStatusApiUrl: string = `${TestWorkspaceUtil.WORKSPACE_API_URL}/${namespace}:${workspaceName}`;
-        const attempts: number = TestConstants.TS_SELENIUM_WORKSPACE_STATUS_ATTEMPTS;
-        const polling: number = TestConstants.TS_SELENIUM_WORKSPACE_STATUS_POLLING;
+        const workspaceStatusApiUrl: string = `${TestWorkspaceUtil.WORKSPACE_API_URL}/${namespace}-che/devworkspaces/${workspaceName}`;
         let workspaceStatus: string = '';
-
-        for (let i = 0; i < attempts; i++) {
+        let expectedStatus: boolean = false;
+        for (let i = 0; i < this.attempts; i++) {
             const response = await this.processRequestHandler.get(workspaceStatusApiUrl);
 
             if (response.status !== 200) {
-                await this.driverHelper.wait(polling);
-                continue;
+                throw new Error(`Can not get status of a workspace. Code: ${response.status} Data: ${response.data}`);
             }
 
-            workspaceStatus = await response.data.status;
+            workspaceStatus = await response.data.status.phase;
 
             if (workspaceStatus === expectedWorkspaceStatus) {
-                return;
+                expectedStatus = true;
+                break;
             }
 
-            await this.driverHelper.wait(polling);
+            await this.driverHelper.wait(this.polling);
+        }
+
+        if (!expectedStatus) {
+            let waitTime = TestConstants.TS_SELENIUM_PLUGIN_PRECENCE_ATTEMPTS * TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+            throw new error.TimeoutError(`The workspace was not stopped in ${waitTime} ms. Currnet status is: ${workspaceStatus}`);
+        }
+    }
+
+    public async stopWorkspaceByName(namespace: string, workspaceName: string) {
+        Logger.debug('TestWorkspaceUtil.stopWorkspaceByName');
+
+        const stopWorkspaceApiUrl: string = `${TestWorkspaceUtil.WORKSPACE_API_URL}/${namespace}-che/devworkspaces/${workspaceName}`;
+        let stopWorkspaceResponse;
+
+        try {
+            stopWorkspaceResponse = await this.processRequestHandler.patch(stopWorkspaceApiUrl, [{'op': 'replace', 'path': '/spec/started', 'value': false}]);
```

Comment:
What about handling 404 error, similar to how it's done in `deleteWorkspaceByName()` method:
```
           } catch (error) {
                if (axios.isAxiosError(error) && error.response?.status === 404) {
                    deleteWorkspaceStatus = true;
                    break;
                }
            }
```